magicctapipe/scripts/lst1_magic/semi_automatic_scripts/coincident_events.py

```diff
@@ -0,0 +1,374 @@
+
+    if (len(LST_runs) == 2) and (len(LST_runs[0]) == 10):
+        LST = LST_runs
+
+        LST_runs = []
+        LST_runs.append(LST)
+
+    if (len(date) == 2) and (len(date[0]) == 10):
+        dt = date
+        date = []
+        date.append(dt)
+    if NSB_match:
+        coincidence_DL1_dir = f"{target_dir}/v{__version__}"
+        if not os.path.exists(f"{coincidence_DL1_dir}/DL1Coincident/"):
+            os.mkdir(f"{coincidence_DL1_dir}/DL1Coincident")
+        for p in ST_list:
+            MAGIC_DL1_dir = f"{target_dir}/v{__version__}/DL1/{p}"
+            if not os.path.exists(f"{coincidence_DL1_dir}/DL1Coincident/{p}"):
+                os.mkdir(f"{coincidence_DL1_dir}/DL1Coincident/{p}")
+            dates = [
+                os.path.basename(x)
+                for x in glob.glob(f"{MAGIC_DL1_dir}/Merged/Merged_*")
+            ]
+            for d in dates:
+                Y_M = int(d.split("_")[1])
+                M_M = int(d.split("_")[2])
+                D_M = int(d.split("_")[3])
+
+                day_MAGIC = dtdt(Y_M, M_M, D_M)
+
+                delta = timedelta(days=1)
+                for i in LST_runs:
+                    Y_L = i[0].split("_")[0]
+                    M_L = i[0].split("_")[1]
+                    D_L = i[0].split("_")[2]
+                    day_LST = dtdt(int(Y_L), int(M_L), int(D_L))
+                    if day_MAGIC == day_LST + delta:
+                        if not os.path.exists(
+                            f"{coincidence_DL1_dir}/DL1Coincident/{p}/NSB{nsb}"
+                        ):
+                            os.mkdir(
+                                f"{coincidence_DL1_dir}/DL1Coincident/{p}/NSB{nsb}"
+                            )
+
+                        lstObsDir = (
+                            i[0].split("_")[0] + i[0].split("_")[1] + i[0].split("_")[2]
+                        )
+
+                        inputdir = f"/fefs/aswg/data/real/DL1/{lstObsDir}/{LST_version}/tailcut84"
+                        if not os.path.exists(
+                            f"{coincidence_DL1_dir}/DL1Coincident/{p}/NSB{nsb}/{lstObsDir}"
+                        ):
+                            os.mkdir(
+                                f"{coincidence_DL1_dir}/DL1Coincident/{p}/NSB{nsb}/{lstObsDir}"
+                            )
+                        if not os.path.exists(
+                            f"{coincidence_DL1_dir}/DL1Coincident/{p}/NSB{nsb}/{lstObsDir}/logs"
+                        ):
+                            os.mkdir(
+                                f"{coincidence_DL1_dir}/DL1Coincident/{p}/NSB{nsb}/{lstObsDir}/logs"
+                            )
+
+                        outputdir = f"{coincidence_DL1_dir}/DL1Coincident/{p}/NSB{nsb}/{lstObsDir}"
+                        list_of_subruns = np.sort(
+                            glob.glob(f"{inputdir}/dl1*Run*{i[1]}*.*.h5")
+                        )
+                        if os.path.exists(f"{outputdir}/logs/list_LST"):
```

Comment:
I think you do not need an if in here. If you simply open the file with 'a' mode, even if it does not exist it will be created, this will simplify the code